iquip/apps/result_explorer.py

```diff
@@ -75,22 +84,99 @@ def run(self):
         self.fetched.emit(ridList)
 
 
+class _H5FileThread(QThread):
+    """QThread for obtaining the H5 format result file from the proxy server.
+    
+    Signals:
+        fetched(results):
+          The H5 format result file is fetched.
+          The "results" is a dictionary with the following keys.
+            expid: The experiment identifier containing submitted information.
+            datasets: A dictionary with datasets.
+              Each key is a dataset name, and its value is a numpy array.
+            submission_time: The submission date-time in ISO format string.
+            start_time: The start date-time in ISO format string.
+            run_time: The run date-time in ISO format string.
+            visualize: True if the visualize option is checked, otherwise False.
+
+    Attributes:
+        rid: The run identifier value of the target executed experiment.
+    """
+
+    fetched = pyqtSignal(dict)
+
+    def __init__(
+        self,
+        rid: str,
+        callback: Callable[[Dict[str, Any]], None],
+        parent: Optional[QObject] = None
+    ):
+        """Extended.
+        
+        Args:
+            rid: See the attributes section in _H5FileThread.
+            callback: The callback method called after this thread is finished.
+        """
+        super().__init__(parent=parent)
+        self.rid = rid
+        self.fetched.connect(callback, type=Qt.QueuedConnection)
+
+    def run(self):
+        """Overridden.
+
+        Fetches the H5 format result file and extracts results.
+
+        After finished, the fetched signal is emitted.
+        """
+        try:
+            response = requests.get(f"http://127.0.0.1:8000/result/{self.rid}/h5/", timeout=10)
+            response.raise_for_status()
+            file_contents = response.content
+            results = {}
+            with h5py.File(io.BytesIO(file_contents), "r") as f:
+                # expid
+                expid_str = f["expid"][()].decode("utf-8")  # pylint: disable=no-member
```

Comment:
I analyzed it more because I didn't know it before you mentioned it.

First, the type of `f["expid"]` is `<HDF5 dataset "expid": shape (), type "|O">`
and that of `f["submission_time"` is `<HDF5 dataset "submission_time": shape (1,), type "|O">`.

So, for "expid" I used `[()]` and for "..._time" I used `[0]`.

But I think I can update the proxy server to store "..._time" into a scalar like "expid" if necessary.
However it isn't responsible for this issue, so please never mind!

Reply:
Thank you for the inspection. So it is just the matter of how the proxy server stores the data (as a scalar or a 1D array), right?

Reply:
Yes, you're perfectly correct!